src/nock/matcher.ts

```diff
@@ -427,16 +432,31 @@ export async function matchTraffic(
 
   if (match.valid) {
     // do not save new recording as there were no breaking changes found
+    // only write recordings with decoded responses if they do not exist
+    if (
+      parseBooleanEnv(process.env.DECODE_RESPONSE) &&
+      !(await exists(oldRecordingDecodedPath))
+    ) {
+      await writeDecodedRecordings(oldRecordingPath, newTraffic);
+    }
+
     return { impact: MatchImpact.NONE };
   } else {
     const impact = analyzeChangeImpact(match.errors);
 
     // Save new traffic
     await writeRecordings(
-      composeRecordingPath(oldRecordingPath, 'new'),
+      composeRecordingPath(oldRecording, { version: 'new' }),
       newTraffic
     );
 
+    // save new traffic with decoded response
+    if (parseBooleanEnv(process.env.DECODE_RESPONSE)) {
+      await writeDecodedRecordings(oldRecordingPath, newTraffic, {
+        version: 'new',
+      });
```

Comment:
Issue: Shouldn't this be:
```suggestion
      await writeDecodedRecordings(oldRecordingPath, newTraffic, {
        version: 'new',
        decoded: true
      });
```

Reply:
The `decoded` parameter is added inside `writeDecodedRecordings` function.

Reply:
🤦 ok I completely missed the function name and since there is separate function to check the existence I confused myself.